src/v/pandaproxy/schema_registry/sharded_store.cc

```diff
@@ -153,7 +153,20 @@ sharded_store::project_ids(subject_schema schema) {
     auto s_id = co_await _store.map_reduce0(
       map, std::optional<schema_id>{}, reduce);
 
-    if (!s_id) {
+    if (schema.id != invalid_schema_id) {
+        if (s_id.has_value() && s_id != schema.id) {
+            co_return ss::coroutine::return_exception(exception(
+              error_code::subject_version_schema_id_already_exists,
+              fmt::format(
+                "Schema already registered with id {} instead of input id {}",
+                s_id.value()(),
+                schema.id())));
+        } else {
+            // Use the supplied id
+            s_id = schema.id;
+            vlog(plog.debug, "project_ids: using supplied ID {}", s_id.value());
+        }
```

Comment:
Based on this:

> The schema ID used for subject-level migration should not already be assigned to a schema in the destination Schema Registry; otherwise the registration of the schema with the same ID will replace the previous schema and any messages serialized with the previous schema will not be able to be deserialized properly.

I think you replace schemas.